x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/LimitedRole.java

```diff
@@ -74,6 +78,29 @@ public RemoteIndicesPermission remoteIndices() {
         throw new UnsupportedOperationException("cannot retrieve remote indices permission on limited role");
     }
 
+    @Override
+    public WorkflowsRestriction workflowsRestriction() {
+        if (false == baseRole.hasWorkflowsRestriction()) {
+            return limitedByRole.workflowsRestriction();
+        }
+        if (false == limitedByRole.hasWorkflowsRestriction()) {
+            return baseRole.workflowsRestriction();
```

Comment:
The implementation of this method can be simplified to just return `baseRole.workflowsRestriction()` since we don't expect that `limitedByRole` has any workflow restriction.